src/cargo/ops/cargo_compile.rs

```diff
@@ -62,6 +62,29 @@ pub struct CompileOptions<'a> {
     pub target_rustc_args: Option<&'a [String]>,
 }
 
+impl<'a> CompileOptions<'a> {
+    pub fn with_default<F, T>(config: &Config, mode: CompileMode, f: F) -> T
```

Comment:
I tried again, and I don't think we can implement Default, because we want to take the config and mode as arguments, but we can just add a `default` method, we can get rid of the taking a closure thing.